SimpleGame/src/main/scala-2.11/edu/gemini/experimental/jluhrs/Main.scala

```diff
@@ -22,18 +29,17 @@ object Main {
       System.out.print("\n" * (height+2*border))
       System.out.flush();
     }
-
     clearScreen
     System.out.print(s)
     System.out.flush();
-
-    //This is just to animate the frames. The real timing must come from the input timing stream.
-    Thread.sleep(500)
   }
 
-  def main(args: Array[String]): Unit =
-    (runGame(width, height, periodicTimer(0.0, 1.0),
-      Stream[Option[Event[PlayerInput]]](Some(5.0, Right), Some(15.0, Up), Some(25.0, Right))
-    ).take(50).toList map (frame(width, height, border, _))).map(displayFrame)
+  def main(args: Array[String]): Unit = {
+
+    ConsoleKeys.start
+
+    (runGame(width, height, (awakeEvery(100 milliseconds).map((_, TimeTick)) merge ConsoleKeys.inputP))
```

Comment:
So this merges the time tick with the console input, nice.
How would it work with the real time?